### PR TITLE
Add graphoid API spec, and hook it up in the example config

### PR DESCRIPTION
**Pull request by gwicke**

### CODE REVIEW

File: config.example.yaml [hunk 1]
```diff
@@ -101,6 +101,18 @@ templates:
             options:
               apiURI: http://{domain}/w/api.php
 
+      /{module:graphoid}:
+        x-modules:
+          - name: simple_service
```
Comment (nyurik) [inline config.example.yaml]: ??

File: specs/mediawiki_graphoid_v1.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+swagger: 2.0
+
+paths:
+  /{module:page}/graph/png/{title}/{revision}/{graph_id}:
+    get:
+      tags:
+        - Page content
+      description: >
+        Retrieve PNG renders of graphs embedded in specific revisions of a
```
Comment (nyurik) [inline specs/mediawiki_graphoid_v1.yaml]: Renders graph images embedded in specific revisions of a

File: specs/mediawiki_graphoid_v1.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          type: integer
+          required: true
+        - name: graph_id
+          in: path
+          description: The graph ID, consisting of a hash and the suffix `.png`.
+          type: string
+          required: true
+      responses:
+        '200':
+          description: The PNG render of the requested graph.
```
Comment (nyurik) [inline specs/mediawiki_graphoid_v1.yaml]: Don't specify PNG - it could switch/add SVG at any point

File: specs/mediawiki_graphoid_v1.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        '404':
+          description: Unknown page title or revision
+          schema:
+            $ref: '#/definitions/problem'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+      x-backend-request:
+        uri: /{domain}/sys/graphoid/v1/png/{title}/{revision}/{graph_id}
```
Comment (nyurik) [inline specs/mediawiki_graphoid_v1.yaml]: /{format}/  instead of /png/   also, graph_id includes .format

File: config.example.yaml (hunk 1 above)
Comment (gwicke) [inline config.example.yaml]: This sets up the internal backend. We could also easily enable storage here; see https://github.com/wikimedia/restbase/blob/master/config.test.yaml#L121-L126 for an example.

File: specs/mediawiki_graphoid_v1.yaml (hunk 3 above)
Comment (gwicke) [inline specs/mediawiki_graphoid_v1.yaml]: That would be a different end point at `/svg/`.

File: specs/mediawiki_graphoid_v1.yaml (hunk 4 above)
Comment (gwicke) [inline specs/mediawiki_graphoid_v1.yaml]: The suffix is part of the ID for our purposes.

File: specs/mediawiki_v1_graphoid.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          type: string
+          required: true
+        - name: revision
+          in: path
+          description: The page revision
+          type: integer
+          required: true
+        - name: graph_id
+          in: path
+          description: The graph ID, consisting of a hash and the suffix `.png`.
```
Comment (d00rman) [inline specs/mediawiki_v1_graphoid.yaml]: *optionally containing the suffix .png* - .png is not required

**Comment by d00rman** [comment]
LGTM :+1:

**Comment by nyurik** [comment]
I think you should replace the word "PNG" with "image" - we should at some point switch to SVGs.

**Comment by d00rman** [comment]
> I think you should replace the word "PNG" with "image" - we should at some point switch to SVGs.

Not really. When SVGs become available, we will simply add another route which will address them exclusively, just like the existing one addresses only PNGs.

File: specs/mediawiki_v1_graphoid.yaml (hunk 5 above)
Comment (nyurik) [inline specs/mediawiki_v1_graphoid.yaml]: i have removed that ability - there should be only one, normalized URL

**Comment by nyurik** [comment]
Why not simply make the /png/ into a parameter /{format}/, just like every other parameter? What's the point of stressing that this is a PNG?

**Comment by gwicke** [comment]
@nyurik, the goal of swagger specs is to document the input-output behavior of the API. This includes the mime type of the response, which in this case is an `image/png`. It supports returning several different types per entry point as well, but it's harder to document the relationship between `{format}` and return value, and in any case at this point PNG is all that's supported. In any case, we can discuss that later as it won't be a breaking change.

**Pull request merged**